hnn_core/network.py

```diff
@@ -509,8 +557,99 @@ def plot_input(self, ax=None, show=True):
             plt.show()
         return ax.get_figure()
 
-    def plot_spikes(self, ax=None, show=True):
-        """Plot the spiking activity for each cell type.
+
+class Spikes(object):
+    '''The Spikes class.
+
+    Parameters
+    ----------
+    times : tuple (n_trials, ) of list of float | None
+        Each element of the tuple is a trial.
+        The list contains the time stamps of spikes.
+    gids : tuple (n_trials, ) of list of float | None
+        Each element of the tuple is a trial.
+        The list contains the cell IDs of neurons that spiked.
+    types : tuple (n_trials, ) of list of float | None
+        Each element of the tuple is a trial.
+        The list contains the type of spike (e.g., evprox1 or
+        L2_pyramidal) that occured at the corresonding time stamp.
+        Each gid corresponds to a type via Network::gid_dict.
+
+    Attributes
+    ----------
+    times : tuple (n_trials, ) of list of float
+        Each element of the tuple is a trial.
+        The list contains the time stamps of spikes.
+    gids : tuple (n_trials, ) of list of float
+        Each element of the tuple is a trial.
+        The list contains the cell IDs of neurons that spiked.
+    types : tuple (n_trials, ) of list of float
+        Each element of the tuple is a trial.
+        The list contains the type of spike (e.g., evprox1 or
+        L2_pyramidal) that occured at the corresonding time stamp.
+        Each gid corresponds to a type via Network::gid_dict.
+
+    Methods
+    -------
+    update_types : update spike types in the current instance of
+        Spikes
+    plot : plot and return a matplotlib Figure object showing the
+        aggregate network spiking activity according to cell type
+    write : write spiking activity to a collection of spike trial
+        files
+    '''
+
+    def __init__(self, times=None, gids=None, types=None):
+        from neuron import h
+
+        self.times = times
```

Comment:
you need to do some input validation. Check that the shapes are consistent etc. Something like [this](https://github.com/mne-tools/mne-python/blob/master/mne/epochs.py#L403-L406)